.github/workflows/deploy.yml

```diff
@@ -0,0 +1,136 @@
+name: Deploy
+
+on:
+  push:
+    tags:
+    - '*.*.*'
+
+jobs:
+  version_check:
+    runs-on: ubuntu-20.04
+    outputs:
+      v_tracker: ${{ steps.version.outputs.RN_TRACKER_VERSION}}
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Setup Node
+      uses: actions/setup-node@v2
+      with:
+        node-version: 14
+
+    - name: Get tag and tracker versions
+      id: version
+      run: |
+        echo ::set-output name=TAG_VERSION::${GITHUB_REF#refs/*/}
+        echo "##[set-output name=RN_TRACKER_VERSION;]$(node -p "require('./package.json').version")"
+
+    - name: Fail if version mismatch
+      if: ${{ steps.version.outputs.TAG_VERSION != steps.version.outputs.RN_TRACKER_VERSION }}
+      run: |
+        echo "Tag version (${{ steps.version.outputs.TAG_VERSION }}) doesn't match version in project (${{ steps.version.outputs.RN_TRACKER_VERSION }})"
+        exit 1
+
+  build_android:
+    needs: ["version_check"]
+    runs-on: ubuntu-20.04
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Validate Gradle Wrapper
+      uses: gradle/wrapper-validation-action@v1
+
+    - name: Setup Node
+      uses: actions/setup-node@v2
+      with:
+        node-version: 14
+
+    - name: Install node_modules
+      working-directory: DemoApp
+      run: |
+        yarn install --frozen-lockfile
+
+    - name: Build DemoApp
+      working-directory: DemoApp/android
+      run: |
+        ./gradlew assembleDebug
+
+  build_ios:
+    needs: ["version_check"]
+    runs-on: macos-10.15
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Setup Node
+      uses: actions/setup-node@v2
+      with:
+        node-version: 14
+
+    - name: Install node_modules
+      working-directory: DemoApp
+      run: |
+        yarn install --frozen-lockfile
+
+    - name: Install Pods
+      working-directory: DemoApp/ios
+      run: |
+        pod install
+
+    - name: Build DemoApp
+      working-directory: DemoApp/ios
+      run: |
+        xcodebuild \
+          -workspace DemoApp.xcworkspace \
+          -scheme DemoApp \
+          -sdk iphonesimulator \
+          -configuration Debug \
+          -destination 'platform=iOS Simulator,name=iPhone 12' \
+          -derivedDataPath ios/build -UseModernBuildSystem=YES \
+          clean build \
+          CODE_SIGNING_ALLOWED=NO
+
+  publish_npm:
+    needs: ["build_android", "build_ios", "version_check"]
+    runs-on: ubuntu-20.04
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Setup Node
+      uses: actions/setup-node@v2
+      with:
+        node-version: 14
+
+    - name: Npm ci
+      run: npm ci
+
+    - name: NPM Publish
+      uses: JS-DevTools/npm-publish@v1
```

Comment:
Any reason why we're using `JS-DevTools/npm-publish@v1` rather than a straight up `npm publish`? I have no issue with this, it looks pretty well maintained, I was just curious what benefits it added?

Reply:
I just liked the features it includes, like the minor version check with npm's latest and the security one (not writing the secret). I am also a bit worried for trying it, so i can certainly replace it with an `npm publish`.

Reply:
No reason to switch it out, it looks like a useful tool. Happy to give it a try and see how it goes, likely allow us to move a little faster than using `npm publish` directly.